tidy3d/components/monitor.py

```diff
@@ -11,6 +11,7 @@
 from .base import cached_property, Tidy3dBaseModel
 from .mode import ModeSpec
 from .apodization import ApodizationSpec
+from .medium import MediumType
```

Comment:
@tylerflex do you think importing from medium here could be problematic in the future? I'm not sure anymore if we ever defined the heirarchy of what's allowed to import from what.

Reply:
Shoot, i don't remember now but this is giving me Deja vu remembering some annoying refactorings. So I'm not sure. My feeling is we probably shouldn't import it here if possible. is there a way around it? Im on mobile and can't look now in detail

Reply:
Bump

Reply:
One idea, to allow for frequency dependence, is a `projection_index` field that accepts either a scalar, or an array whose size is validated to match the size of `monitor.freqs`, so that the user can provide a constant or one value per frequency?

Reply:
That only enables this feature for client-side projections - I think for server side, we need to modify `AbstractFieldProjectionMonitor` anyway, because the solver is unaware of `FieldProjector`

Reply:
At one point we had discussed getting rid of client-side projections altogether, especially if the plan is to provide a standalone server-side projector. Is this still the the goal? If so, maybe the clunkiness is tolerable since it's mostly `field_projection.py` that's clunky now?

Reply:
It also seems more annoying for the user if it's not taking medium directly. Let's go with medium.